datasets/squad_checklist/squad_checklist.py

```diff
@@ -0,0 +1,112 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""A CheckList suite for SQuAD"""
+
+from __future__ import absolute_import, division, print_function
+
+import csv
+import json
+import logging
+import os
+
+import nlp
+from checklist.test_suite import TestSuite
+
+
+_CITATION = """\
+ @inproceedings{checklist:acl20},
+ author = {Marco Tulio Ribeiro and Tongshuang Wu and Carlos Guestrin and Sameer Singh},
+ title = {Beyond Accuracy: Behavioral Testing of NLP models with CheckList},
+ booktitle = {Association for Computational Linguistics (ACL)},
+ year = {2020}
+"""
+
+_DESCRIPTION = """\
+A CheckList for SQuAD.
+Predictions: each prediction is a string, containing the answer
+Confidences: not necessary for this checklist
+
+Test names for Table 3 in the paper:
+['A is COMP than B. Who is more / less COMP?', 'Intensifiers (very, super, extremely) and reducers (somewhat, kinda, etc)?', 'size, shape, age, color', 'Profession vs nationality', 'Animal vs Vehicle v2', 'A is COMP than B. Who is antonym(COMP)? B', 'A is more X than B. Who is more antonym(X)? B. Who is less X? B. Who is more X? A. Who is less antonym(X)? A.', 'Question typo', 'Add random sentence to context', 'There was a change in profession', 'Understanding before / after -> first / last.', 'Negation in context, may or may not be in question', 'Negation in question only.', 'M/F failure rates should be similar for different professions', 'Basic coref, he / she', 'Basic coref, his / her', 'Former / Latter', 'Agent / object distinction', 'Agent / object distinction with 3 agents']
+
+Use with nlp.checklist.CheckListSuite
+"""
+
+_URL = "https://github.com/marcotcr/checklist/raw/master/release_suites/squad_suite.tar.gz"
+_SUITE_NAME = "squad_suite.pkl"
+
+
+class SquadCheckListConfig(nlp.BuilderConfig):
+    """ BuilderConfig for NewDataset"""
+
+    def __init__(self, url=_URL, suite_name=_SUITE_NAME, **kwargs):
+        """
+
+        Args:
+            url: .tar.gz file url
+            suite_name: name of the pickle file archived in url
+            **kwargs: keyword arguments forwarded to super.
+        """
+        self.url = url
+        self.suite_name = suite_name
+        super(SquadCheckListConfig, self).__init__(**kwargs)
+
+
+class SquadCheckList(nlp.GeneratorBasedBuilder):
+    """TODO: Short description of my dataset."""
+
+    VERSION = nlp.Version("1.1.0")
+
+    BUILDER_CONFIG_CLASS = SquadCheckListConfig
+    BUILDER_CONFIGS = [SquadCheckListConfig(_URL, _SUITE_NAME, version=VERSION, name="squad_checklist_config")]
+
```

Comment:
```suggestion

    test_dummy_data = False

```

Reply:
Do we really want to have them untested ?

Reply:
The alternative is to install `checklists` for the CI, are we OK with that?

Reply:
Also, looks like that doesn't actually fix the tests as the class is still loaded (and raises the import error).

Do we have any other datasets with optional dependencies @lhoestq ? How did we handle that?

Reply:
Let's add it to the `tests` section of the dependencies in `setup.py`. This is how we do it for other datasets with external reps.

Reply:
I tried @thomwolf 's suggestion but the CI version of python seems to be missing the `notebook` dependency which is required in the `checklist` `setup.py`. Is there an easy way around that?